compiler/damlc/daml-compiler/src/DA/Daml/Compiler/DataDependencies.hs

```diff
@@ -838,25 +838,31 @@ isSuperClassField (LF.FieldName fieldName) =
     "s_" `T.isPrefixOf` fieldName
 
 -- | Signature data for a dictionary function.
-data DFunSig
-    = DFunSigNormal -- ^ signature for a normal dictionary function
-        { dfsType :: !LF.Type -- ^ LF type signature for the dictionary function
-        }
-    | DFunSigHasField -- ^ signature for a HasField dictionary function
-        { dfsBinders :: ![(LF.TypeVarName, LF.Kind)] -- ^ foralls
-        , dfsContext :: ![LF.Type] -- ^ constraints
-        , dfsName :: LF.Qualified LF.TypeSynName -- ^ name of type synonym
-        , dfsField :: !T.Text -- ^ first arg (a type level string)
-        , dfsArgs :: ![LF.Type] -- ^ rest of the args
-        }
+data DFunSig = DFunSig
+    { dfsBinders :: ![(LF.TypeVarName, LF.Kind)] -- ^ foralls
+    , dfsContext :: ![LF.Type] -- ^ constraints
+    , dfsHead :: !DFunHead
+    }
+
+-- | Instance declaration head
+data DFunHead
+    = DFunHeadHasField -- ^ HasField instance
+          { dfsName :: LF.Qualified LF.TypeSynName -- ^ name of type synonym
+          , dfsField :: !T.Text -- ^ first arg (a type level string)
+          , dfsArgs :: [LF.Type] -- ^ rest of the args
+          }
+    | DFunHeadNormal -- ^ Normal, i.e., non-HasField, instance
+          { dfsName :: LF.Qualified LF.TypeSynName -- ^ name of type synonym
+          , dfsArgs :: [LF.Type] -- ^ arguments
```

Comment:
Might be nice to rename these fields to `dfhName`, etc

Reply:
Good point, thanks!